prometheus/vec.go

```diff
@@ -216,7 +228,7 @@ func (m *MetricVec) hashLabelValues(vals []string) (uint64, error) {
 	}
 	h := hashNew()
 	for _, val := range vals {
-		h = hashAdd(h, val)
+		h = m.hashAdd(h, val)
```

Comment:
This should actually be followed by `hashAddByte(h, separatorByte)` (this is a bug in the original code already, hash collisions could be created with label values like `{"foo", "bar"}` and `{"foob", "ar"}`).